src/index.ts

```diff
@@ -678,192 +511,82 @@ class NixInstallerAction extends DetSysAction {
+          let msg: string;
+          if (daemon.signalCode) {
+            msg = `Daemon was killed by signal ${daemon.signalCode}`;
+          } else {
+            msg = `Daemon exited with code ${daemon.exitCode}`;
+          }
+
+          throw new Error(msg);
         }
-      }
 
-      const plausibleDeterminateOptions = [];
-      const plausibleDeterminateArguments = [];
-      if (this.determinate) {
-        plausibleDeterminateOptions.push("--entrypoint");
-        plausibleDeterminateOptions.push("/usr/local/bin/determinate-nixd");
-        plausibleDeterminateArguments.push("daemon");
-      }
-
-      this.recordEvent(EVENT_START_DOCKER_SHIM);
-      const exitCode = await actionsExec.exec(
-        "docker",
-        [
-          "--log-level=debug",
-          "run",
-          "--detach",
-          "--privileged",
-          "--network=host",
-          "--userns=host",
-          "--pid=host",
-          "--restart",
-          "always",
-          "--init",
-          "--name",
-          `determinate-nix-shim-${this.getUniqueId()}-${randomUUID()}`,
-        ]
-          .concat(plausibleDeterminateOptions)
-          .concat(mountArguments)
-          .concat(["determinate-nix-shim:latest"])
-          .concat(plausibleDeterminateArguments),
-        {
-          silent: true,
-          listeners: {
-            stdline: (data: string) => {
-              actionsCore.saveState("docker_shim_container_id", data.trimEnd());
-            },
-            stdout: (data: Buffer) => {
-              const trimmed = data.toString("utf-8").trimEnd();
-              if (trimmed.length >= 0) {
-                actionsCore.debug(trimmed);
-              }
-            },
-            stderr: (data: Buffer) => {
-              const trimmed = data.toString("utf-8").trimEnd();
-              if (trimmed.length >= 0) {
-                actionsCore.debug(trimmed);
-              }
-            },
-          },
-        },
-      );
+        if (await this.doesTheSocketExistYet()) {
+          break;
+        }
 
-      if (exitCode !== 0) {
-        throw new Error(
-          `Failed to start the Nix daemon through Docker, exit code: \`${exitCode}\``,
-        );
+        await setTimeout(50);
       }
-    }
-
-    const maxDurationSeconds = 120;
-    const delayPerAttemptInMiliseconds = 50;
-    const maxAttempts =
-      (maxDurationSeconds * 1000) / delayPerAttemptInMiliseconds;
-    let didSucceed = false;
 
-    for (let attempt = 0; attempt <= maxAttempts; attempt += 1) {
-      if (await this.doesTheSocketExistYet()) {
-        didSucceed = true;
-        break;
+      if (!(await this.doesTheSocketExistYet())) {
+        throw new Error("Timed out waiting for the daemon socket to appear.");
       }
+    } catch (error: unknown) {
+      this.recordEvent(EVENT_NO_SYSTEMD_SHIM_FAILED, {
+        error: stringifyError(error),
+        log: (await readFile(outputPath)).toString(),
```

Comment:
This seems fishy. Do you want to read the file with an encoding?

```suggestion
        log: await readFile(outputPath, "utf-8"),
```